### PR TITLE
Adding AdaMod Optimizer in Tensorflow 2.x

### PR DESCRIPTION
**Pull request by monk1337**
**System information**

TensorFlow version (you are using): 2.0
Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

I am looking to add one new optimizer called "Adamod" . Adamod was  proposed by Jianbang Ding in the paper "An Adaptive and Momental Bound Method for Stochastic Learning"
AdaMod uses a new parameter called “Beta3”. The results are improved convergence, no need for warmup, and less sensitivity to the actual learning rate chosen. Beta3 parameter controls the degree of memory.

**Will this change the current api? How?**
I 've edited `optimizers.py` and created a new file adamod.py which is compatible with Tensorflow 2.x.
Usage :

```
model.compile(optimizer='adamod',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
```

### CODE REVIEW

**Comment by tanzhenyu** [comment]
Thanks for the PR!
Have you checked with developers@tensorflow.org about having a new optimizer? Usually if it's not widely used we would recommend to put it under tensorflow addons first.

**Comment by gbaned** [comment]
@monk1337 Can you please check reviewer comments and keep us posted. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
